data/extension/init/templates/tinygo/envoy.filters.network/default/main_test.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxytest"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func newStreamContext(uint32) proxywasm.StreamContext {
+	return context{}
+}
+
+func TestNetwork_OnNewConnection(t *testing.T) {
+	host, done := proxytest.NewNetworkFilterHost(newStreamContext)
+	defer done() // release the host emulation lock so that other test cases can insert their own host emulation
+
+	_ = host.InitConnection() // OnNewConnection is called
+
+	logs := host.GetLogs(types.LogLevelInfo) // retrieve logs emitted to Envoy
+	assert.Equal(t, logs[0], "new connection!")
+}
+
+func TestNetwork_counter(t *testing.T) {
+	host, done := proxytest.NewNetworkFilterHost(newStreamContext)
+	defer done() // release the host emulation lock so that other test cases can insert their own host emulation
+
+	context{}.OnVMStart(0) // init metric
```

Comment:
> IIUC, context corresponds to a "(Network) Stream" in terms of Proxy Wasm.

I think the word `context` is used not only for (network)streams but also for vm/plugin which we call RootContext.

But I know this looks strange and this is because I should use `StartVM` function in proxytest/* emulation instead.